src/viam/app/data_client.py

```diff
@@ -478,6 +495,89 @@ async def get_database_connection(self, organization_id: str) -> str:
     async def configure_database_user(self, organization_id: str, password: str) -> None:
         raise NotImplementedError()
 
+    async def create_dataset(self, name: str, organization_id: str) -> str:
+        """Create a new dataset.
+
+        Args:
+            name (str): The name of the dataset being created.
+            organization_id (str): The ID of the organization where the dataset is being created.
+
+        Returns:
+            str: The dataset ID of the created dataset.
+        """
+        request = CreateDatasetRequest(name=name, organization_id=organization_id)
+        response: CreateDatasetResponse = await self._dataset_client.CreateDataset(request, metadata=self._metadata)
+        return response.id
+
+    async def list_dataset_by_ids(self, ids: List[str]) -> List[Dataset]:
+        """Get a list of datasets using their IDs.
+
+        Args:
+            ids (List[str]): The IDs of the datasets being called for.
+
+        Returns:
+            List[Dataset]: The list of datasets.
+        """
+        request = ListDatasetsByIDsRequest(ids=ids)
+        response: ListDatasetsByIDsResponse = await self._dataset_client.ListDatasetsByIDs(request, metadata=self._metadata)
+
+        return list(response.datasets)
+
+    async def list_datasets_by_organization_id(self, organization_id: str) -> List[Dataset]:
+        """Get the datasets in an organization.
+
+        Args:
+            organization_id (str): The ID of the organization.
+
+        Returns:
+            List[Dataset]: The list of datasets in the organization.
+        """
+        request = ListDatasetsByOrganizationIDRequest(organization_id=organization_id)
+        response: ListDatasetsByOrganizationIDResponse = await self._dataset_client.ListDatasetsByOrganizationID(
+            request, metadata=self._metadata
+        )
+
+        return list(response.datasets)
+
+    async def rename_dataset(self, id: str, name: str) -> None:
+        """Rename a dataset specified by the dataset ID.
+
+        Args:
+            id (str): The ID of the dataset.
+            name (str): The new name of the dataset.
+        """
+        request = RenameDatasetRequest(id=id, name=name)
+        await self._dataset_client.RenameDataset(request, metadata=self._metadata)
+
+    async def delete_dataset(self, id: str) -> None:
+        """Delete a dataset.
+
+        Args:
+            id (str): The ID of the dataset.
+        """
+        request = DeleteDatasetRequest(id=id)
+        await self._dataset_client.DeleteDataset(request, metadata=self._metadata)
+
+    async def add_binary_data_to_dataset_by_ids(self, binary_ids: List[BinaryID], dataset_id: str) -> None:
+        """Add VIAM_DATASET_{id} tag to binary data.
```

Comment:
Perhaps for this method and the one below it (remove tag), the docstring should start more high level --

> Add the `BinaryData` to the provided `DataSet`.
>
> This `BinaryData` will be tagged with the `VIAM_DATASET_{id}` label